### PR TITLE
Fix/1440 not able to select second pm

### PR DESCRIPTION
**Pull request by seanbrookes**
connected to #1165 
connected to #1360 
connected to #1439 
connected to #1443 
connected to #1460 
connected to #1440

### CODE REVIEW

**Comment by seanbrookes** [comment]
test please

File: client/www/scripts/modules/metrics/metrics.controllers.js [hunk 1]
```diff
@@ -43,15 +57,20 @@ Metrics.controller('MetricsMainController', [
...
       $scope.processes = processes;
-      $scope.updateProcessSelection([processes[0]]);
+      if (processes.map) {
+        $scope.updateProcessSelection([processes[0]]);
+      }
     };
 
     $scope.updateProcessSelection = function(selection) {
-      if (selection.length) {
+      if (selection.length && selection[0]) {
```
Comment (anthonyettinger) [inline client/www/scripts/modules/metrics/metrics.controllers.js]: Seems redundant to test `selection[0]` here.

**Comment by anthonyettinger** [comment]
1 comment, otherwise LGTM.

May want to have @jtary take a look too.

File: client/www/scripts/modules/pm/pm.services.js [hunk 2]
```diff
@@ -398,7 +412,7 @@ PM.service('PMServerService', ['$http', '$log',
           baseUrl = '/process-manager'
         }
         else if (serverConfig.port === PM_CONST.LOCAL_PM_PORT_MASK){
-          $log.warn('invalid port - may be corruped request: ' + JSON.stringify(serverConfig));
+          $log.warn('invalid port - may be corupted request: ' + JSON.stringify(serverConfig));
```
Comment (jtary) [inline client/www/scripts/modules/pm/pm.services.js]: typo here

**Comment by jtary** [comment]
Minor typo, but other than that LGTM

**Comment by seanbrookes** [comment]
test please

**Comment by seanbrookes** [comment]
test please

**Comment by seanbrookes** [comment]
test please

**Pull request merged**